### PR TITLE
Changed pattern matching on json results to jq, removed writing log

### PR DESCRIPTION
**Pull request by georgr**
pattern matching on json results didn´t work reliable; also removed text file writing as it is wearing out the router flash memory

### CODE REVIEW

File: cloudflare-ddns.sh [hunk 1]
```diff
@@ -3,57 +3,56 @@
...
 
 # CHANGE THESE
-auth_email="mail@hogehoge.com"            # The email used to login 'https://dash.cloudflare.com'
-auth_key="xxxxx"   # Top right corner, "My profile" > "Global API Key"
-zone_identifier="xxxxx" # Can be found in the "Overview" tab of your domain
-record_name="fuga.hoge.com"                     # Which record you want to be synced
+auth_email="mail@example.com"           # The email used to login 'https://dash.cloudflare.com'
+auth_key="xxxxx"   			# Top right corner, "My profile" > "Global API Key"
+zone_identifier="xxxxx" 		# Can be found in the "Overview" tab of your domain
+record_name="home.examlpe.com"          # Which record you want to be synced
```
Comment (mu373) [inline cloudflare-ddns.sh:9]: typo: example

File: cloudflare-ddns.sh [hunk 2]
```diff
@@ -3,57 +3,56 @@
...
 # Update IP address through Cloudflare API
 update=$(curl -s -X PUT "https://api.cloudflare.com/client/v4/zones/$zone_identifier/dns_records/$record_identifier" \
-	-H "X-Auth-Email: $auth_email" \
-	-H "X-Auth-Key: $auth_key" \
-	-H "Content-Type: application/json" \
-	--data "{\"type\":\"A\",\"name\":\"$record_name\",\"content\":\"$ip\",\"ttl\":1}")
-
-# Log IP address if it has changed
-currenttime=$(date '+%Y-%m-%dT%H:%M%z')
-echo "$currenttime, $ip" >> ip-address.txt
```
Comment (mu373) [inline cloudflare-ddns.sh:48]: As I prefer keeping logs, maybe there can be an option for this. In my environment, IP address does not change that often and does not take much volume.

```bash
keep_log=true # at the top config part
if "${keep_log}"; then
	currenttime=$(date '+%Y-%m-%dT%H:%M%z')	
	echo "$currenttime, $ip" >> ip-address.txt
fi
```

File: cloudflare-ddns.sh [hunk 3]
```diff
@@ -3,57 +3,56 @@
...
-# Fetch Global IP address
+# Fetch external IP address
 ip=$(curl -s inet-ip.info)
 
 # Seek for the current record on Cloudflare
 record=$(curl -s -X GET "https://api.cloudflare.com/client/v4/zones/$zone_identifier/dns_records?name=$record_name" \
-	-H "X-Auth-Email: $auth_email" \
-	-H "X-Auth-Key: $auth_key" \
-	-H "Content-Type: application/json")
+        -H "X-Auth-Email: $auth_email" \
```
Comment (mu373) [inline cloudflare-ddns.sh:16]: Although it's my personal preference, I prefer using tabs to spaces. (Same for all indents throughout the script)

**Review by mu373 — CHANGES_REQUESTED** [review]
Thank you for the pull request!
I agree that `jq` is more reliable to extract data from JSON, even when there are data specification changes. I initially used `grep` because I wanted to make this script work in other environments where `jq` is not installed.

I would merge this request once the feedbacks on comments are addressed.

File: README.md [hunk 4]
```diff
@@ -10,4 +10,34 @@ This script is intended to be used for specific needs:
...
+
+Copy the script with (win)scp to /config/scripts of your router.
+
+Make it executable: `sudo chmod 0755 /config/scripts/cloudflare-ddns.sh`
+
+Schedule task:
+```
+configure
+set system task-scheduler task cloudflare-ddns executable path /config/scripts/cloudflare-ddns.sh
+set system task-scheduler task cloudflare-ddns interval 1m
```
Comment (mu373) [inline README.md:29]: I feel that 1 min interval is too much. When considering burden on external APIs, I think that the default value should be longer (i.e. 15min)

File: README.md [hunk 5]
```diff
@@ -10,4 +10,34 @@ This script is intended to be used for specific needs:
...
+show system task-scheduler
+commit-confirm
+confirm
+save
```
Comment (mu373) [inline README.md:33]: add `exit`

Comment (mu373) [inline README.md:33]: add `exit`

File: README.md [hunk 6]
```diff
@@ -10,4 +10,34 @@ This script is intended to be used for specific needs:
...
+Fill out in the header of the script
+
+- auth_email: The email used to login 'https://dash.cloudflare.com
+- auth_key: Top right corner, "My profile" > "Global API Key"
+- zone_identifier: Can be found in the "Overview" tab of your domain on the right side on the lower half of the page
+- record_name: Domain on which you want to update the A record
+
+Copy the script with (win)scp to /config/scripts of your router.
+
+Make it executable: `sudo chmod 0755 /config/scripts/cloudflare-ddns.sh`
```
Comment (mu373) [inline README.md:23]: remove `0` from `0755`

```bash
sudo chmod 755 /config/scripts/cloudflare-ddns.sh
```

**Review by mu373 — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (georgr) [reply README.md:29]: Would it help to poll DNS except the API (e.g. dig +short @8.8.8.8), 15 minutes to detect a change seems a bit long to me.

**Pull request closed without merging**